src/mock/parsec/mod.rs

```diff
@@ -293,6 +299,10 @@ where
             Observation::Add { ref peer_id, .. } => self.peer_list.insert(peer_id.clone()),
             Observation::Remove { ref peer_id, .. } => self.peer_list.remove(peer_id),
             Observation::Accusation { ref offender, .. } => self.peer_list.remove(offender),
+            Observation::StartDkg(ref participants) => {
+                self.dkg_participants.extend(participants.iter().cloned());
+                false
+            }
```

Comment:
The set of participants is only ever extended - this would be wrong if a Parsec instance could survive a membership change. I'd add a comment somewhere that we don't need to remove participants after DKG is complete, because the Parsec instance will be abandoned soon, anyway.

Reply:
Parsec does not remove participants after DKG is complete either, but it does do so when peer are removed, so that is what I did in the new version.